src/kansei/mpu.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2021 Ichiro ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <kansei/mpu.hpp>
+
+#include <keisan/angle.hpp>
+
+#include <cmath>
+#include <string>
+
+#include "./fcntl.h"
+#include "./termios.h"
+#include "./stdio.h"
+#include "./unistd.h"
+#include "./string.h"
+#include "./errno.h"
+#include "sys/stat.h"
+#include "sys/ioctl.h"
+
+#define MINIMUM_ERROR 0.001
+
+namespace kansei
+{
+
+MPU::MPU(const std::string & port_name)
+: angle_(0), angle_error_(0)
+{
+  set_port_name(port_name);
+}
+
+MPU::~MPU()
+{
+  if (socket_fd_ != -1) {
+    close(socket_fd_);
+  }
+  socket_fd_ = -1;
+}
+
+bool MPU::connect()
+{
+  socket_fd_ = open(serial_name_.c_str(), O_RDWR | O_NOCTTY);
+  if (socket_fd_ > 0) {
+    angle_ = 0;
+    angle_error_ = 0;
+    angle_compensation_ = 0;
+    calibrated_ = false;
+
+    int iFlags = TIOCM_DTR;
+    ioctl(socket_fd_, TIOCMBIS, &iFlags);
+
+    iFlags = TIOCM_DTR;
+    ioctl(socket_fd_, TIOCMBIC, &iFlags);
+
+    struct termios newtio;
+    bzero(&newtio, sizeof(newtio));
+
+    newtio.c_cflag = B115200 | CRTSCTS | CS8 | CLOCAL | CREAD;
+    newtio.c_iflag = IGNPAR | ICRNL;
+    newtio.c_oflag = 0;
+    newtio.c_lflag = ~ICANON;
+
+    newtio.c_cc[VINTR] = 0;
+    newtio.c_cc[VQUIT] = 0;
+    newtio.c_cc[VERASE] = 0;
+    newtio.c_cc[VKILL] = 0;
+    newtio.c_cc[VEOF] = 4;
+    newtio.c_cc[VTIME] = 0;
+    newtio.c_cc[VMIN] = 1;
+    newtio.c_cc[VSWTC] = 0;
+    newtio.c_cc[VSTART] = 0;
+    newtio.c_cc[VSTOP] = 0;
+    newtio.c_cc[VSUSP] = 0;
+    newtio.c_cc[VEOL] = 0;
+    newtio.c_cc[VREPRINT] = 0;
+    newtio.c_cc[VDISCARD] = 0;
+    newtio.c_cc[VWERASE] = 0;
+    newtio.c_cc[VLNEXT] = 0;
+    newtio.c_cc[VEOL2] = 0;
```

Comment:
`newtio` is already cleared with `0`. it's unnecessary to set some part of `newtio` to `0`. Also, it's better to give comment on why the `c_lflag` must be set to `~ICANON`, why `c_cc[VMIN]` must be set to `1`, etc. Feel free to remove unnecessary things, or add something that must be added.